service/src/cli.yml

```diff
@@ -26,20 +26,32 @@ args:
         default_value: "9944"
     - ws-external:
         long: ws-external
-        help: let worker api listen to external requests too      
+        help: Set this flag in case the worker should listen to external requests.
     - mu-ra-port:
         short: r
         long: mu-ra-port
         help: Set the websocket port to listen for mu-ra requests
         takes_value: true
         default_value: "3443"
-    - worker-rpc-port:
+    - trusted-worker-port:
         short: P
-        long: worker-rpc-port
-        help: Set the websocket port where the worker rpc direct invocations port listens
+        long: trusted-worker-port
+        help: Set the trusted websocket port of the worker, running directly in the enclave.
         takes_value: true
         default_value: "2000"
-    
+    - untrusted-worker-port:
+        short: U
+        long: untrusted-worker-port
+        help: Set the untrusted websocket port of the worker
+        takes_value: true
+        default_value: "2001"
+    - external-address:
+        short: ext
+        long: external-address
+        help: Set the worker's address to be advertised on the parentchain. The port will be the same as given in `trusted-worker-port`
```

Comment:
Hmm, we can't always use the `trusted-worker-port` here because if we run the worker behind an nginx instance, the external port will be different from the trusted-worker-port. Does it still default to 443 if no port is given?

Reply:
I honestly don't know from where that 443  is coming from. Never seen that value except in this comment.
Whatever was the case before, all we have now is:  https://github.com/integritee-network/worker/blob/master/service/src/main.rs#L147

Can we agree that I will remove the port input in case the external address is set  (i.e. if external address is set, port input is neccessary!) and add a feature issue for making port input optional for external address?

Reply:
Added this functionality, hence resolving this conversation.